### DISCUSSION TITLE
variable gradient (_grad_u[_qp]) along one direction

### DISCUSSION THREAD

**Discussion by Bearunner** [idaholab/moose#26587] [Q&A General]
Dear all,

We use **_grad_u[_qp]** to represent the gradient of a variable along all directions in a 3D model.

How could we use variable gradient along one direction, e.g., x direction in the kernel?

Thanks.

B

**Comment by GiudGiud** [comment]
Hello

You would index into the vector (`_grad_u[_qp]`) to retrieve one component

Guillaume

**Reply by Bearunner** [reply]
like _grad_u[_qp][0]?

**Reply by Bearunner** [reply]
if (u-T1)/dx is the gradient along the x direction and T1 is a coupled value from other input, how can we write it in kernels?

**Reply by GiudGiud** [reply]
From other input? What does you mean?

```
_grad_u[_qp](_component);
```
not []

**Reply by Bearunner** [reply]
Thanks for your answer.


lambda*∇((u-Tres.)/dx).  
where Tres. is a coupled value temperature from the reservoir.

**Reply by GiudGiud** [reply]
∇((u-Tres.)/dx). ?
do you mean 
 div (∇((u-Tres.) )
in continuous form ?
or do you mean that you are trying to do some 1D finite differencing?
(∇u- ∇Tres)  / dx
